arcgis-ios-sdk-samples/Layers/Raster layer (file)/RasterLayerVC.swift

```diff
@@ -42,15 +42,11 @@ class RasterLayerVC: UIViewController {
         
         self.mapView.map?.operationalLayers.add(rasterLayer!)
         
-        self.rasterLayer.addObserver(self, forKeyPath: "loadStatus", options: .new, context: nil)
-        
-    }
-
-    
-    override func observeValue(forKeyPath keyPath: String?, of object: Any?, change: [NSKeyValueChangeKey : Any]?, context: UnsafeMutableRawPointer?) {
-        if self.rasterLayer.loadStatus == AGSLoadStatus.loaded {
-            self.mapView.setViewpoint(AGSViewpoint(center: (self.rasterLayer.fullExtent?.center)!, scale: 80000))
-        }
+        self.rasterLayer.load { (error) in
+            if error == nil {
+                self.mapView.setViewpoint(AGSViewpoint(center: (self.rasterLayer.fullExtent?.center)!, scale: 80000))
```

Comment:
Add weak reference to self and update this line of code to remove implicit unwrap. Use optional binding instead.